packages/crypto/benches/main.rs

```diff
@@ -0,0 +1,138 @@
+
+const COSMOS_ED25519_MSG: &str = "";
+const COSMOS_ED25519_SIGNATURE_HEX: &str = "e5564300c360ac729086e2cc806e828a84877f1eb8e5d974d873e065224901555fb8821590a33bacc61e39701cf9b46bd25bf5f0595bbe24655141438e7a100b";
+const COSMOS_ED25519_PUBLIC_KEY_HEX: &str =
+    "d75a980182b10ab7d54bfed3c964073a0ee172f3daa62325af021a68f707511a";
+
+// Test data from https://tools.ietf.org/html/rfc8032#section-7.1
+const COSMOS_ED25519_TESTS_JSON: &str = "./testdata/ed25519_tests.json";
+
+#[derive(Deserialize, Debug)]
+struct Encoded {
+    #[serde(rename = "privkey")]
+    private_key: String,
+    #[serde(rename = "pubkey")]
+    public_key: String,
+    message: String,
+    signature: String,
+}
+
+fn read_cosmos_sigs() -> Vec<Encoded> {
+    use std::fs::File;
+    use std::io::BufReader;
+
+    // Open the file in read-only mode with buffer.
+    let file = File::open(COSMOS_ED25519_TESTS_JSON).unwrap();
+    let reader = BufReader::new(file);
+
+    serde_json::from_reader(reader).unwrap()
+}
+
+fn bench_crypto(c: &mut Criterion) {
+    let mut group = c.benchmark_group("Crypto");
+
+    group.bench_function("secp256k1_verify", |b| {
+        let message = hex::decode(COSMOS_SECP256K1_MSG_HEX).unwrap();
+        let message_hash = Sha256::digest(&message);
+        let signature = hex::decode(COSMOS_SECP256K1_SIGNATURE_HEX).unwrap();
+        let public_key = base64::decode(COSMOS_SECP256K1_PUBKEY_BASE64).unwrap();
+        b.iter(|| {
+            assert!(secp256k1_verify(&message_hash, &signature, &public_key).unwrap());
+        });
+    });
+
+    group.bench_function("secp256k1_recover_pubkey", |b| {
+        let message_hash =
+            hex!("82ff40c0a986c6a5cfad4ddf4c3aa6996f1a7837f9c398e17e5de5cbd5a12b28");
+        let private_key =
+            hex!("3c9229289a6125f7fdf1885a77bb12c37a8d3b4962d936f7e3084dece32a3ca1");
+        let r_s = hex!("99e71a99cb2270b8cac5254f9e99b6210c6c10224a1579cf389ef88b20a1abe9129ff05af364204442bdb53ab6f18a99ab48acc9326fa689f228040429e3ca66");
+        let recovery_param: u8 = 0;
+
+        let expected = SigningKey::from_bytes(&private_key)
+            .unwrap()
+            .verify_key()
+            .to_encoded_point(false)
+            .as_bytes()
+            .to_vec();
+
+        b.iter(|| {
+            let pubkey = secp256k1_recover_pubkey(&message_hash, &r_s, recovery_param).unwrap();
+            assert_eq!(pubkey, expected);
+        });
+    });
+
+    group.bench_function("ed25519_verify", |b| {
+        let message = COSMOS_ED25519_MSG.as_bytes();
+        let signature = hex::decode(COSMOS_ED25519_SIGNATURE_HEX).unwrap();
+        let public_key = hex::decode(COSMOS_ED25519_PUBLIC_KEY_HEX).unwrap();
+        b.iter(|| {
+            assert!(ed25519_verify(&message, &signature, &public_key).unwrap());
+        });
+    });
+
+    // Batch verification of five entries
+    group.bench_function("ed25519_batch_verify_five", |b| {
```

Comment:
Could we loop over a few numbers here (e.g. 1, 5, 10, 20, 50)?

Reply:
Sure. These would be n different `bench_function`s, right?

How to get 20 different test vectors, by the way. I can clone / repeat them, but that would distort the benchmark.

Reply:
>  These would be n different bench_functions, right?

Right

> How to get 20 different test vectors, by the way. I can clone / repeat them, but that would distort the benchmarks.

Ah, I did not check where the test data comes from. Then let's use 1, 3, 5, which gives us the structure and we can extend later.

Reply:
I'll do 1,2,3,4,5, which will give us some data points to extrapolate. We can always extend it later.